Integration tests fail with java9

The maven integration tests fail with java9 (See below):

Note also that it is great to have these IT tests as they found a bug in a merge that I otherwise would not have noticed.  However, it is not clear how to run them, nor how to look at their output, nor how to look at them in an IDE.

Can we add an README_INTEGRATION_TEST.md file somewhere that describes:
 + the profile needed to run the tests
 + how the test source can be imported into an IDE
 + the output files from the tests eg `target/it/jetty-run-distro-mojo-it/build.log` and `target/it/jetty-run-distro-mojo-it/jetty-simple-webapp/target/jetty.out`

Perhaps even reference these files in error messages when a test fails?


```
[INFO] --- maven-invoker-plugin:3.0.1:integration-test (integration-test) @ jetty-maven-plugin ---
[INFO] Running Setup Jobs
[INFO] Building: jetty-run-distro-mojo-it/pom.xml
[INFO] run post-build script postbuild.groovy
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/home/gregw/.m2/repository/org/codehaus/groovy/groovy-all/2.4.8/groovy-all-2.4.8.jar) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[INFO]           jetty-run-distro-mojo-it/pom.xml ................. SUCCESS (12.4 s)
[INFO] Building: jetty-run-war-exploded-mojo-it/pom.xml
[INFO]           jetty-run-war-exploded-mojo-it/pom.xml ........... FAILED (5.9 s)
[INFO]   The build exited with code 1. See /usr/local/home/gregw/src/jetty-9.4.x/jetty-maven-plugin/target/it/jetty-run-war-exploded-mojo-it/build.log for details.
[INFO] Building: jetty-start-mojo-it/pom.xml
[INFO]           jetty-start-mojo-it/pom.xml ...................... SUCCESS (4.8 s)
[INFO] Building: jetty-run-war-mojo-it/pom.xml
[INFO]           jetty-run-war-mojo-it/pom.xml .................... FAILED (4.6 s)
[INFO]   The build exited with code 1. See /usr/local/home/gregw/src/jetty-9.4.x/jetty-maven-plugin/target/it/jetty-run-war-mojo-it/build.log for details.
[INFO] Building: jetty-run-mojo-it/pom.xml
[INFO]           jetty-run-mojo-it/pom.xml ........................ SUCCESS (5.5 s)
[INFO] Building: jetty-run-forked-mojo-it/pom.xml
[INFO] run post-build script postbuild.groovy
[INFO]           jetty-run-forked-mojo-it/pom.xml ................. SUCCESS (5.8 s)
[INFO] 
[INFO] --- maven-invoker-plugin:3.0.1:verify (integration-test) @ jetty-maven-plugin ---
[INFO] -------------------------------------------------
[INFO] Build Summary:
[INFO]   Passed: 4, Failed: 2, Errors: 0, Skipped: 0
[INFO] -------------------------------------------------
[ERROR] The following builds failed:
[ERROR] *  jetty-run-war-mojo-it/pom.xml
[ERROR] *  jetty-run-war-exploded-mojo-it/pom.xml
[INFO] -------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 49.011 s
[INFO] Finished at: 2018-01-31T11:59:42+01:00
[INFO] Final Memory: 47M/157M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-invoker-plugin:3.0.1:verify (integration-test) on project jetty-maven-plugin: 2 builds failed. See console output above for details. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

--- a/jetty-maven-plugin/README_INTEGRATION_TEST.md
+++ b/jetty-maven-plugin/README_INTEGRATION_TEST.md
@@ -1,0 +1,24 @@
+Running Maven Integration tests
+=====================
+The project contains a set of Maven Integration test projects.
+They are running using the Maven Invoker plugin which starts an external Maven build to run the project and some post build check.
+More details [http://maven.apache.org/plugins/maven-invoker-plugin/].
+
+Integration tests location
+--------------------
+Test projects are located within the folder: src/it
+
+Running the tests
+--------------------
+As they can be long to run, the tests do not run per default. So to run them you must activate a profile using the command line argument: ```-Prun-its``` 
+
+Running single test
+--------------------
+You can run single or set of test as well using the command line argument: ```-Dinvoker.test=jetty-run-mojo-it,jetty-run-war*-it,!jetty-run-distro*```
+The parameter supports pattern and exclusion with !
+
+Running Logs
+--------------------
+The output of each Maven build will be located in /target/it/${project-name}/build.log
+
+

--- a/jetty-maven-plugin/pom.xml
+++ b/jetty-maven-plugin/pom.xml
@@ -15,7 +15,7 @@
     <bundle-symbolic-name>${project.groupId}.maven.plugin</bundle-symbolic-name>
     <surefireVersion>2.20.1</surefireVersion>
     <it.debug>false</it.debug>
-    <jetty.stopKey>FOOBEER</jetty.stopKey>
+    <jetty.stopKey>FREEBEER</jetty.stopKey>
     <jetty.jvmArgs></jetty.jvmArgs>
   </properties>
   <build>

--- a/jetty-maven-plugin/pom.xml
+++ b/jetty-maven-plugin/pom.xml
@@ -13,6 +13,7 @@
     <mavenVersion>3.5.0</mavenVersion>
     <pluginToolsVersion>3.5</pluginToolsVersion>
     <bundle-symbolic-name>${project.groupId}.maven.plugin</bundle-symbolic-name>
+    <surefireVersion>2.20.1</surefireVersion>
     <it.debug>false</it.debug>
     <jetty.stopKey>FOOBEER</jetty.stopKey>
     <jetty.jvmArgs></jetty.jvmArgs>
@@ -224,6 +225,7 @@
                 <jettyStopKey>${jetty.stopKey}</jettyStopKey>
                 <jettyStopPort>${jetty.stopPort}</jettyStopPort>
                 <jettyRunPort>${jetty.runPort}</jettyRunPort>
+                <surefireVersion>${surefireVersion}</surefireVersion>
               </scriptVariables>
               <goals>
                 <goal>clean</goal>

--- a/jetty-maven-plugin/src/it/jetty-run-war-exploded-mojo-it/invoker.properties
+++ b/jetty-maven-plugin/src/it/jetty-run-war-exploded-mojo-it/invoker.properties
@@ -1,2 +1,2 @@
-invoker.goals = verify
+invoker.goals = verify -V
 #test-compile failsafe:integration-test

--- a/jetty-maven-plugin/src/it/jetty-run-war-exploded-mojo-it/jetty-simple-webapp/pom.xml
+++ b/jetty-maven-plugin/src/it/jetty-run-war-exploded-mojo-it/jetty-simple-webapp/pom.xml
@@ -64,7 +64,7 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-surefire-plugin</artifactId>
-        <version>2.20</version>
+        <version>@surefireVersion@</version>
         <configuration>
           <skip>true</skip>
           <systemPropertyVariables>
@@ -75,7 +75,7 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-failsafe-plugin</artifactId>
-        <version>2.20</version>
+        <version>@surefireVersion@</version>
         <configuration>
           <systemPropertyVariables>
             <jetty.port.file>${jetty.port.file}</jetty.port.file>

--- a/jetty-maven-plugin/src/it/jetty-run-war-mojo-it/invoker.properties
+++ b/jetty-maven-plugin/src/it/jetty-run-war-mojo-it/invoker.properties
@@ -1,2 +1,2 @@
-invoker.goals = verify
+invoker.goals = verify -V
 #test-compile failsafe:integration-test

--- a/jetty-maven-plugin/src/it/jetty-run-war-mojo-it/jetty-simple-webapp/pom.xml
+++ b/jetty-maven-plugin/src/it/jetty-run-war-mojo-it/jetty-simple-webapp/pom.xml
@@ -66,7 +66,7 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-surefire-plugin</artifactId>
-        <version>2.20</version>
+        <version>@surefireVersion@</version>
         <configuration>
           <skip>true</skip>
           <systemPropertyVariables>
@@ -77,7 +77,7 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-failsafe-plugin</artifactId>
-        <version>2.20</version>
+        <version>@surefireVersion@</version>
         <configuration>
           <systemPropertyVariables>
             <jetty.port.file>${jetty.port.file}</jetty.port.file>

